src/components/Search.astro

```diff
@@ -0,0 +1,263 @@
+    private isSearchLoaded = false;
+
+    constructor() {
+      super();
+      this.searchButton = this.querySelector("#search-button");
+
+      if (!this.searchButton) {
+        console.error("Search button not found");
+        return;
+      }
+
+      // Create modal when component initializes
+      this.createModal();
+
+      // Setup event listeners
+      this.searchButton.addEventListener("click", this.openSearch.bind(this));
+    }
+
+    createModal() {
+      // Check if modal already exists
+      if (document.getElementById("search-modal")) return;
+
+      // Create modal HTML
+      const modal = document.createElement("div");
+      modal.id = "search-modal";
+      modal.className = "fixed inset-0 bg-black bg-opacity-50 z-50 hidden";
+      modal.setAttribute("role", "dialog");
+      modal.setAttribute("aria-modal", "true");
+      modal.setAttribute("aria-labelledby", "search-title");
+      modal.innerHTML = `
+        <div class="flex items-start justify-center pt-16 px-4 pb-20 text-center sm:block sm:p-0">
+          <div class="inline-block align-bottom bg-white dark:bg-gray-800 rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full">
+            <!-- Modal header -->
+            <div class="bg-white dark:bg-gray-800 px-4 pt-5 pb-4 sm:p-6 sm:pb-4">
+              <div class="flex items-center justify-between mb-4">
+                <h2 id="search-title" class="text-lg font-medium text-gray-900 dark:text-gray-100">Search</h2>
+                <button 
+                  id="search-close" 
+                  class="text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-500 rounded-md"
+                  aria-label="Close search"
+                >
+                  <svg class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12" />
+                  </svg>
+                </button>
+              </div>
+              
+              <!-- Pagefind search container -->
+              <div id="pagefind-search" class="w-full"></div>
+            </div>
+          </div>
+        </div>
+      `;
+
+      // Append to body
+      document.body.appendChild(modal);
+
+      // Setup modal event listeners
+      const closeButton = modal.querySelector("#search-close");
+      closeButton?.addEventListener("click", this.closeSearch.bind(this));
+      
+      // Close modal when clicking outside
+      modal.addEventListener("click", (e) => {
+        if (e.target === modal) {
+          this.closeSearch();
+        }
+      });
+
+      // Close modal with Escape key
+      document.addEventListener("keydown", (e) => {
+        if (e.key === "Escape" && !modal.classList.contains("hidden")) {
+          this.closeSearch();
+        }
+      });
+    }
+
+    async openSearch() {
```

Comment:
The search display "Search is temporarily unavailable. Please try again later" and doesn't allow the user to type anything, what is wrong?